src/modules/preferences.js

```diff
@@ -89,6 +89,7 @@ PassFF.Preferences = (function () {
     showStatus            : true,
     tbMenuShortcut        : '',
     recognisedSuffixes    : 'co.uk,org.uk,me.uk,co.jp,com.au',
+    filterFieldRegex      : 'notes,att((ributes)|(achments))',
```

Comment:
I don't know why, the regex `notes|att((ributes)|(achments))` with 'gi' (global, case insensitive) doesn't match consistently the words "attachments" and "notes"

```javascript
var reg = new RegExp('notes|att((ributes)|(achments))', 'gi')
let truthy = [ "notes", "attributes", "attachments" ];
let falsy = [ "not", "at", "web", "pass", "notations", "att" ];

falsy.filter(e => reg.test(e)); 
// expected output: [ ]
// actual output: [ ]

truthy.filter(e => !reg.test(e));
// expected output: [ ]
// actual output: [ "notes" ] or [ "notes", "attachments" ]
```
Unsetting the global flag `g` (which seems useless anyway?) fixes the problem.

Reply:
Good catch, thanks. Not familiar enough with regexes to tell you why that happens, either, but I'll remove the global flag. If you can provide a convincing enough reason to remove the case-insensitive flag, I'll do so as well -- however, I'd rather `attachments` and `Attachments` both match, in case a user capitalizes a letter by accident.

Reply:
From the description of `test`:

```
If the regex has the global flag set, test() will advance the lastIndex of the regex. A subsequent use of test() will start the search at the substring of str specified by lastIndex (exec() will also advance the lastIndex property). It is worth noting that the lastIndex will not reset when testing a different string.
```

The method `test` mutates the Regex, and since you aren't reinstantiating it for each iteration of the loop, you are getting mixed results. Your test is, therefore, the problem, and not the regex.

Regardless, the `g` flag is inconsequential for the `isHidden` method, since we only case about finding a single match. Therefore, it is unnecessary.

Additionally, a proper regex engine will do the work of compiling `notes|attributes|attachments` into a DFA which will be equivalent to writing `notes|att(ributes|achments)` because it will collapse the common prefixes. Therefore, for clarity, I would recommend making the default regex `notes|attributes|attachments` for improved clarity.